### ISSUE TITLE
OrderedDict.fromList doesn't remove duplicate keys from the key order

### ISSUE THREAD

**Issue by mkaszas** [j-maas/ordered-containers#22]
When using the `fromList` function, the `order` list can end up having duplicate keys.

```
> OrderedDict.fromList [("a", 1), ("a", 2)]
OrderedDict { dict = Dict.fromList [("a",2)], order = ["a","a"] }
```

This is especially dangerous when getting the values later:
```
> OrderedDict.fromList [("a", 1), ("a", 2)] |> OrderedDict.values
[2,2]
```

I would expect the above to behave like inserting:
```
> OrderedDict.singleton "a" 1 |> OrderedDict.insert "a" 2
OrderedDict { dict = Dict.fromList [("a",2)], order = ["a"] }
```

A good solution would probably be to run `nub` on the order list when building the `OrderedDict`, or fold the list with the `insert` function?

**Comment by j-maas** [comment]
That is indeed a serious bug. Thanks for the great issue description!

~The only thing I want to figure out before I release a new version is what the best procedure is. Right now I leaning towards publishing a new major version, since this is technically a breaking change. But I'll ask on the Elm slack what others would recommend.~

On second thought this could be a patch release, since that is the intended behavior. I'll still double check on the Elm slack.

**Comment by j-maas** [comment]
I've published the fix as version 2.0.2.

**Comment by mkaszas** [comment]
Amazing! Thank you for the quick response, and for maintaining this package! 🎉
